src/main/java/RedisSessionManagerT8.java

```diff
@@ -0,0 +1,298 @@
+    }
+
+    //------------------------------------------------------------------------------------------------------------------
+    // START Sessions
+    //------------------------------------------------------------------------------------------------------------------
+
+    @Override
+    public int getRejectedSessions() {
+        // Essentially do nothing.
+        return 0;
+    }
+
+    @Override
+    public Session createSession(String requestedSessionId) {
+
+        Session session = createEmptySession();
+
+        // Initialize the properties of the new session and return it
+        session.setId(UUID.randomUUID().toString());
+        session.setNew(true);
+        session.setValid(true);
+        session.setCreationTime(System.currentTimeMillis());
+        session.setMaxInactiveInterval(this.maxInactiveInterval);
+
+        try {
+            Jedis jedis = redisConnectionPool.getResource();
+            saveToRedis(jedis, session);
+            redisConnectionPool.returnResourceObject(jedis);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        return session;
+    }
+
+    /**
+     * Add session to this manager
+     * @param session
+     */
+    @Override
+    public void add(Session session) {
+        try {
+            Jedis jedis = redisConnectionPool.getResource();
+            saveToRedis(jedis, session);
+            redisConnectionPool.returnResourceObject(jedis);
+        } catch (IOException ex) {
+            log.warn("Unable to add to session manager store: " + ex.getMessage());
+            throw new RuntimeException("Unable to add to session manager store.", ex);
+        }
+    }
+
+    @Override
+    public Session findSession(String id) throws IOException {
+        Session session = null;
+
+        //Get connection to redis
+        Jedis jedis = redisConnectionPool.getResource();
+
+        Hashtable<String, Object> metadata =
+                (Hashtable)SerializationUtils.deserialize(Base64.getDecoder().decode(jedis.get((id + REDIS_METADATA_KEY).getBytes())));
+
+        Hashtable<String, Object> attributes =
+                (Hashtable)SerializationUtils.deserialize(Base64.getDecoder().decode(jedis.get((id + REDIS_ATTRIBUTES_KEY).getBytes())));
+
+        session = getSession(metadata, attributes);
+
+        redisConnectionPool.returnResourceObject(jedis);
+
+        return session;
+    }
+
+    @Override
+    public void remove(Session session) {
+
+        remove(session, false);
+    }
+
+    @Override
+    public void remove(Session session, boolean update) {
+        //Get connection to redis
+        Jedis jedis = redisConnectionPool.getResource();
+
+        //Just remove the attributes not the metadata
+        jedis.del((session.getId() + REDIS_ATTRIBUTES_KEY).getBytes());
+
+        redisConnectionPool.returnResourceObject(jedis);
+    }
+
+    @Override
+    public void processExpires() {
+       //We need to implement the expire process
```

Comment:
This should be handled by Redis. You should simply set the expiration time on the attributes.